### ISSUE TITLE
dependencies in jupyter notebooks go unchecked?

### ISSUE THREAD

**Issue by gwdekker** [gauge-sh/tach#492]
Hey, it seems that dependencies in jupyter notebooks do not get checked by tach, is that correct? Seems like a bug if this is correct.

**Comment by emdoyle** [comment]
@gwdekker This is correct. Right now, Tach expects that it will be checking imports in Python source files only, and this assumption exists throughout our other commands.

The core use case we have built Tach for is Python application development, and we have relatively little visibility into how folks might use it with Jupyter Notebooks. Could you say more about your use case?

Functionally, it seems like we should be able to support most of Tach's functionality on notebooks by just condensing all of the Python code, but AFAIK features like [interfaces](https://docs.gauge.sh/usage/interfaces) wouldn't make sense since you cannot import from a notebook. There might also be complexity when we translate the filesystem path to a module path (naming conflicts seem possible if a sibling Python file shares the name of a notebook), but this seems solvable too.

**Comment by gwdekker** [comment]
Sure! My code base is a monorepo which hosts several applications. It als contains a few jupyter notebooks, some of them for ad-hoc analysis, some of them we use through Voila as an ad-hoc dashboard, some of them on Voila even work as a very basic UI to our application. So the notebooks are just as much part of my modules as plain python files are. Therefore, the expected behavior for me would be that import statements in jupyter notebooks would be treated the same way as import statements in plain python files.

Potentially an easy way to achieve this would be to hook in `nbconvert --to script` into your system somehow?

**Comment by mfarrugi** [comment]
@gwdekker you could use jupytext to do this.

**Comment by gwdekker** [comment]
well I am aware of `jupytext`, but in my monorepo its not only me working. Expecting everybody to install `jupytext` in however they work with notebooks, and expecting them to learn how to use it, is a large ask. I am really surprised that notebooks do not "just work" in tach - for me application development in a data science environment, where a repo contains mostly code but also the odd notebook here and there is not a very strange usecase.
